src/util.ts

```diff
@@ -25,14 +25,25 @@ export const defaultOptions: HoneycombOptions = {
 export const createHoneycombSDKLogMessage = (message: string) =>
   `@honeycombio/opentelemetry-web: ${message}`;
 
+const classicKeyRegex = /^[a-f0-9]*$/;
+const ingestClassicKeyRegex = /^hc[a-z]ic_[a-z0-9]*$/;
+
 /**
- * Determines whether the passed in apikey is classic (32 chars) or not.
+ * Determines whether the passed in apikey is classic or not.
  *
  * @param apikey the apikey
  * @returns a boolean to indicate if the apikey was a classic key
  */
 export function isClassic(apikey?: string): boolean {
-  return apikey?.length === 32;
+  if (apikey === null || apikey === undefined || apikey.length === 0) {
+    return false;
+  }
```

Comment:
This behavior is slightly different from how we implemented the logic in [libhoney-js](https://github.com/honeycombio/libhoney-js/pull/406/files#diff-02ac5abd26d8410a1b28c76bb9e40a78226c3a991553b4bafeba363f81dd4a61R228-R230) -- an empty / undefined key will evaluate as _not_ classic, while in libhoney-js and other libhoneys it would evaluate to true. There were explicit tests for this already so I chose to match what was already here, but I'm open to making these match if that's desired.

Reply:
Yeah I think it would've been a breaking change for libhoney otherwise